lib/puppet/type/interface_properties.rb

```diff
@@ -15,43 +15,100 @@
 #
 
 Puppet::Type.newtype(:interface_properties) do
-    @doc = "Manage Oracle Solaris interface properties"
+  @doc = "Manage Oracle Solaris interface properties
+      Protocol must be defined either at the interface/resource name or in
+      the properties hash.
 
-    ensurable do
-        # remove the ability to specify :absent.  New values must be set.
-        newvalue(:present) do
-            provider.create
-        end
+        Preferred:
+        name: net0
+        Properties:
+        A complex hash of proto => { property => value },...
+        {
+          'ipv4' => { 'mtu' => '1776' },
+          'ipv6' => { 'mtu' => '2048' },
+        }
+
+        Old Syntax:
+        name: net0/ipv4
+        Properties:
+        A hash of property => value when Interface defines protocol
+        { 'mtu' => '1776' }
+  "
+
+  ensurable do
+    # remove the ability to specify :absent.  New values must be set.
+    newvalue(:present) do
+      # Do nothing
+    end
+    newvalue(:absent) do
+      # We can only ensure synchronization
+      fail "Interface properties cannot be removed"
     end
+  end
 
-    newparam(:interface) do
-        desc "The name of the interface with protocol (if appropriate)"
-        validate do |iface|
-            if not iface =~ /^.*?\/.*?$/i
-                raise Puppet::Error, "Invalid interface name.  Interface name
-                    must specify name and protocol."
-            end
-        end
-        isnamevar
+  newparam(:name) do
+    desc "The name of the interface"
+    newvalues(/^[a-z_0-9]+[0-9]+(?:\/ipv[46]?)?$/)
+    isnamevar
```

Comment:
We really only want interface here but support ipv[46] as a backwards compatibility option.